### PR TITLE
Merge in actions/cache@v3 changes

### PR DESCRIPTION
**Pull request by dan-blanchard**
I wanted to use this action, but was getting lots of warnings about it using deprecated commands like `set-state` and `save-output`, so I forked it and merged in the changes from upstream.

I then resolved all the merge conflicts, rebuilt, and did the same for your https://github.com/whywaita/actions-toolkit-s3 fork as well in https://github.com/whywaita/actions-toolkit-s3/pull/10

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,31 +23,31 @@
   "author": "GitHub",
   "license": "MIT",
   "dependencies": {
-    "@actions/cache": "https://gitpkg.now.sh/whywaita/toolkit/packages/cache?d423a21",
-    "@actions/core": "^1.2.6",
-    "@actions/exec": "^1.0.1",
-    "@actions/io": "^1.1.0",
+    "@actions/cache": "https://gitpkg.now.sh/dan-blanchard/actions-toolkit-s3/packages/cache?e16f854",
```
Comment (whywaita) [inline package.json]: I had merged [actions-toolkit-s3](https://github.com/whywaita/actions-toolkit-s3/pull/10). Please update package.json.

**Review by whywaita — CHANGES_REQUESTED** [review]

**Comment by whywaita** [comment]
Thank you for your great work!

File: package.json (hunk 1 above)
Comment (dan-blanchard) [reply package.json]: Okay, I've got that set in 6d956d0

**Comment by iglu-sebastian** [comment]
Great to see some work done on this front - to upgrade fully can we update to `node20` ?

**Comment by enchorb** [comment]
When will this be released?

**Comment by whywaita** [comment]
@dan-blanchard Thank you!

I released your patch at https://github.com/whywaita/actions-cache-s3/releases/tag/v3-s3 .

**Pull request merged**